python_modules/dagster/dagster/_core/storage/asset_check_execution_record.py

```diff
@@ -101,3 +121,91 @@ def from_db_row(cls, row) -> "AssetCheckExecutionRecord":
+            # Check if the run is still active, and if not, return the actual status.
+            run_record = RunRecord.blocking_get(loading_context, self.run_id)
+            if not run_record:
+                # Run deleted
+                return AssetCheckExecutionResolvedStatus.SKIPPED
+
+            run = run_record.dagster_run
+            if run.is_finished:
+                return (
+                    AssetCheckExecutionResolvedStatus.EXECUTION_FAILED
+                    if run.status == DagsterRunStatus.FAILURE
+                    else AssetCheckExecutionResolvedStatus.SKIPPED
+                )
+            else:
+                return AssetCheckExecutionResolvedStatus.IN_PROGRESS
+        else:
+            check.failed(f"Unexpected status {self.status}")
+
+    def targets_latest_materialization(self, loading_context: LoadingContext) -> bool:
+        from dagster._core.storage.event_log.base import AssetRecord
+
+        resolved_status = self.resolve_status(loading_context)
+        if resolved_status == AssetCheckExecutionResolvedStatus.IN_PROGRESS:
+            # all in-progress checks execute against the latest version
+            return True
+
+        asset_record = AssetRecord.blocking_get(loading_context, self.key.asset_key)
+        latest_materialization = (
+            asset_record.asset_entry.last_materialization_record if asset_record else None
+        )
+        if not latest_materialization:
+            # no previous materialization, so it's executing against the lastest version
+            return True
+
+        latest_materialization_run_id = latest_materialization.event_log_entry.run_id
+        if latest_materialization_run_id == self.run_id:
+            # part of the same run
+            return True
+
+        if resolved_status in [
+            AssetCheckExecutionResolvedStatus.SUCCEEDED,
+            AssetCheckExecutionResolvedStatus.FAILED,
+        ]:
+            evaluation = check.not_none(self.evaluation)
+            if not evaluation.target_materialization_data:
+                # check ran before the materialization was created
+                return False
+            else:
+                # check matches the latest materialization id
+                return (
+                    evaluation.target_materialization_data.storage_id
+                    == latest_materialization.storage_id
+                )
+        elif resolved_status in [
+            AssetCheckExecutionResolvedStatus.EXECUTION_FAILED,
+            AssetCheckExecutionResolvedStatus.SKIPPED,
+        ]:
+            # the evaluation didn't complete, so we don't have target_materialization_data, so check if
+            # the check's run executed after the materializations as a fallback
+            latest_materialization_run_record = RunRecord.blocking_get(
+                loading_context, latest_materialization_run_id
+            )
+            check_run_record = RunRecord.blocking_get(loading_context, self.run_id)
```

Comment:
still wonder if we need some way to prevent the prefetch -> get pairings from getting disconnected silently